components/profile.js

```diff
@@ -65,57 +67,413 @@ SteamCommunity.prototype.editProfile = function(settings, callback) {
+										out.push({
+											"name": "rgShowcaseConfig[17][" + n + "][title]",
+											"value": defaultval[1]
+										});
+									}
+								}
+								break;
+
+								case 'games':
+									out.push({
+										"name": "profile_showcase[]",
+										"value": "2"
+									});
+
+									if (settings[i][t]["values"].hasOwnProperty("games")) {
+										showcaseconfig = [true, 4, 2, settings[i][t]["values"]["games"]];
+									}
+									break;
+
+								case 'ownguides':
+									out.push({
+										"name": "profile_showcase[]",
+										"value": "16"
+									});
+
+									for (var n = 0; n < 4; n++) {
+										var defaultval = ["", ""];
+										if (settings[i][t]["values"][n] != undefined) {
+											defaultval = [settings[i][t]["values"][n]["appid"], settings[i][t]["values"][n]["publishedfileid"]];
+										}
+										out.push({
+											"name": "rgShowcaseConfig[16][" + n + "][appid]",
+											"value": defaultval[0]
+										});
+										out.push({
+											"name": "rgShowcaseConfig[16][" + n + "][publishedfileid]",
+											"value": defaultval[1]
+										});
+									}
+									break;
+
+								case 'ownguides':
+									out.push({
+										"name": "profile_showcase[]",
+										"value": "12"
+									});
+
+									for (var n = 0; n < 5; n++) {
+										var defaultval = ["", ""];
+										if (settings[i][t]["values"][n] != undefined) {
+											defaultval = [settings[i][t]["values"][n]["appid"], settings[i][t]["values"][n]["publishedfileid"]];
+										}
+										out.push({
+											"name": "rgShowcaseConfig[12][" + n + "][appid]",
+											"value": defaultval[0]
+										});
+										out.push({
+											"name": "rgShowcaseConfig[12][" + n + "][publishedfileid]",
+											"value": defaultval[1]
+										});
+									}
+									break;
+							}
+
+							if (showcaseconfig[0]) {
+								for (var n = 0; n < showcaseconfig[1]; n++) {
+									var requestdata;
+									if (showcaseconfig[3][n] == undefined) {
+										requestdata = {
+											appid: 0,
+											item_contextid: 0,
+											item_assetid: 0,
+											customization_type: showcaseconfig[2],
+											slot: n,
+											sessionid: formd[0].value
+										};
+									} else {
+										requestdata = {
+											appid: showcaseconfig[3][n]["appid"],
+											item_contextid: showcaseconfig[3][n]["item_contextid"],
+											item_assetid: showcaseconfig[3][n]["item_assetid"],
+											customization_type: showcaseconfig[2],
+											slot: n,
+											sessionid: formd[0].value
+										};
+
+									}
+
+									self._myProfile("ajaxsetshowcaseconfig", requestdata, function (err, response, body) {
+										if (settings.customURL) {
+											delete selfe._profileURL;
+										}
+
+										if (err || response.statusCode != 200) {
+											if (callback) {
+												callback(err || new Error("HTTP error " + response.statusCode));
```

Comment:
I'm not sure about this, but it's a `callback` in `for` loop - this will lead to it being possibly fired multiple times.

Reply:
Like I said before in the comments, this is actually of a private project, so I copied the code which was used before in the library to make the requests to edit the profile. I didn't meddle with it and didn't have any repetitive calls when I was debugging my use case.

Reply:
But it should be possible when setting certain showcases. If that's the case, it won't take long until we see first "my bot is <doing something> n times" issues...